### PR TITLE
Add support for Python 3.7 and 3.8 with allowed failure of 3.8

### PR DESCRIPTION
**Pull request by stevepiercy**

### CODE REVIEW

**Comment by stevepiercy** [comment]
See #3312. WIP DO NOT MERGE.

**Comment by stevepiercy** [comment]
The tests passed earlier, so I expect this to be OK to merge, pending this round of tests and peer review.

Closes #3312.

File: docs/quick_tutorial/requirements.rst [hunk 1]
```diff
@@ -19,11 +19,11 @@ virtual environment.)
...
 This *Quick Tutorial* is based on:
 
-* **Python 3.6**. Pyramid fully supports Python 3.4+ and Python 2.7+. This
-  tutorial uses **Python 3.6** but runs fine under Python 2.7.
+* **Python 3.7**. Pyramid fully supports Python 3.4+ and Python 2.7+. This
+  tutorial uses **Python 3.7** but runs fine under Python 2.7.
 
 * **venv**. We believe in virtual environments. For this tutorial, we use
-  Python 3.6's built-in solution :term:`venv`. For Python 2.7, you can install
+  Python 3.7's built-in solution :term:`venv`. For Python 2.7, you can install
```
Comment (mmerickel) [inline docs/quick_tutorial/requirements.rst]: This sentence could just say "Python 3's built-in solution"

File: docs/quick_tour.rst [hunk 2]
```diff
@@ -26,7 +26,7 @@ To save a little bit of typing and to be certain that we use the modules,
 scripts, and packages installed in our virtual environment, we'll set an
 environment variable, too.
 
-As an example, for Python 3.6+ on Linux:
+As an example, for Python 3.7+ on Linux:
```
Comment (mmerickel) [inline docs/quick_tour.rst]: Similarly this could say just "for Python 3 on Linux"

**Comment by mmerickel** [comment]
Sure

**Pull request merged**